connection.go

```diff
@@ -146,6 +149,13 @@ type Connection struct {
 	lenbuf  [PacketLengthBytes]byte
 
 	lastStreamId uint64
+
+	// protocolVersion is IProto max protocol version supported both by
+	// client and server. Equal to ProtocolVersionUnsupported is server
```

Comment:
> max protocol version supported both by client and server

What is the idea? How can this be used in practice?
I find it more useful to know the version of the server.

Reply:
There is no known usage for protocol version yet.

> I find it more useful to know the version of the server.

Well, in case server has newer version than client the only thing we could do is to understand that there are some features we're not know about yet.

Reply:
> Well, in case server has newer version than client the only thing we could do is to understand that there are some features we do'nt know about yet.

It looks better than nothing.

Reply:
> > Well, protocol version is 0 for Tarantool between "version when IPROTO_ID was introduced" and "somewhere between 2.10.0-beta1 and 2.10.0-beta2 when first feature flag was added", so "unknown version" and "version 0" defines different version of Tarantool. On the other hand, it's not like we would treat `-1` and `0` protocol version different in out code.
> 
> No, it is not true, see:
> 
> [tarantool/tarantool@0be1faf1fe](https://github.com/tarantool/tarantool/commit/0be1faf1fe)
> 
> It was introduced with `1`.

Sorry for the misinformation then. I'll use `0` as `ProtocolVersionUnsupported`.

Reply:
Reworked the connector to store server provided info and a handle to check if a feature (or a protocol version) is supported by current connection (server + client).

Reply:
Close this branch since the discussion continues in https://github.com/tarantool/go-tarantool/pull/226#discussion_r1021682159